flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -222,6 +226,20 @@ public int getPageSize() {
 		}
 	}
 
+	public int getNumberOfQueuedBuffers() {
+		int totalBuffers = 0;
+
+		for (Map.Entry<IntermediateResultPartitionID, InputChannel> entry: inputChannels.entrySet()) {
```

Comment:
I think this map may change asynchronously, so it is probably better to catch exceptions here and re-try for some times. It it fails three times, return `-1` for "unknown"